### PR TITLE
[SPARK-11963][DOC] Add docs for QuantileDiscretizer

### PR DESCRIPTION
**Pull request by yinxusen**
https://issues.apache.org/jira/browse/SPARK-11963

### CODE REVIEW

**Comment by jkbradley** [comment]
I'll review now

File: examples/src/main/scala/org/apache/spark/examples/ml/QuantileDiscretizerExample.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+object QuantileDiscretizerExample {
+  def main(args: Array[String]) {
+    val conf = new SparkConf().setAppName("QuantileDiscretizerExample")
+    val sc = new SparkContext(conf)
+    val sqlContext = new SQLContext(sc)
+    import sqlContext.implicits._
+
+    // $example on$
+    val data = Array((0, 18.0), (1, 19.0), (2, 8.0), (3, 5.0), (4, 2.2))
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/ml/QuantileDiscretizerExample.scala]: In the last instance, this uses 2.2, but your Markdown example uses 22.  This must be the cause of the buckets being out-of-order in the example.

**Comment by jkbradley** [comment]
That's the only issue I see.

**Comment by yinxusen** [comment]
Thanks @jkbradley, it's my fault. I tested both JavaQuantileDiscretizer and QuantileDiscretizer just now. Given the 4th row 2.2, we get the result showed in the doc, other than 22.0. So I changed the 22.0 -> 2.2 in both ml-features and JavaQuantileDiscretizerExample.

File: docs/ml-features.md [hunk 2]
```diff
@@ -1705,6 +1705,71 @@ print(output.select("features", "clicked").first())
...
+## QuantileDiscretizer
+
+`QuantileDiscretizer` takes a column with continuous features and outputs a column with binned
+categorical features.
+The bin ranges are chosen by taking a sample of the data and dividing it into roughly equal parts.
+The lower and upper bin bounds will be `-Infinity` and `+Infinity`, covering all real values.
+This attempts to find numBuckets partitions based on a sample of the given input data, but it may
+find fewer depending on the data sample values.
+
+Note that the result may different every time you run it, since the sample strategy behind it is
```
Comment (BenFradet) [inline docs/ml-features.md]: "may be different" or "may differ" but not "may different"

File: docs/ml-features.md [hunk 3]
```diff
@@ -1705,6 +1705,71 @@ print(output.select("features", "clicked").first())
 </div>
 </div>
 
+## QuantileDiscretizer
+
+`QuantileDiscretizer` takes a column with continuous features and outputs a column with binned
+categorical features.
+The bin ranges are chosen by taking a sample of the data and dividing it into roughly equal parts.
+The lower and upper bin bounds will be `-Infinity` and `+Infinity`, covering all real values.
+This attempts to find numBuckets partitions based on a sample of the given input data, but it may
```
Comment (BenFradet) [inline docs/ml-features.md]: I'd put numBuckets between backticks.

**Comment by BenFradet** [comment]
There is a typo but otherwise it looks good.

**Comment by yinxusen** [comment]
Thanks @BenFradet for the comments. I fixed them.

**Comment by jkbradley** [comment]
Thanks!  I'll test once more and then merge

**Comment by jkbradley** [comment]
LGTM, merging with master and branch-1.6

**Pull request closed without merging**